ci-operator/step-registry/baremetalds/sno/rename/baremetalds-sno-rename-commands.sh

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Fetch packet basic configuration
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+
+cat >"${SHARED_DIR}"/recert-rename-sno.sh << "EOF"
+#!/usr/bin/env bash
+
+export PREVIOUS_CLUSTER_NAME="${PREVIOUS_CLUSTER_NAME:-test-infra-cluster}"
+export PREVIOUS_BASE_DOMAIN="${PREVIOUS_BASE_DOMAIN:-redhat.com}"
+export CLUSTER_NAME="${CLUSTER_NAME:-another-name}"
+export BASE_DOMAIN="${BASE_DOMAIN:-another.domain}"
+export SINGLE_NODE_IP="${SINGLE_NODE_IP:-192.168.127.10}"
+
+# assisted-test-infra sets up 2 network interfaces that compete with each other
+# when setting the NODE_IP and KUBELET_NODEIP. Use KUBELET_NODEIP_HINT to
+# ensure the correct interface is chosen.
+#
+# https://access.redhat.com/articles/6956852
+ssh -o UserKnownHostsFile=/dev/null \
+  -o StrictHostKeyChecking=no \
+  -o LogLevel=ERROR \
+  "core@${SINGLE_NODE_IP}" \
+  "echo 'KUBELET_NODEIP_HINT=192.168.127.1' | sudo tee /etc/default/nodeip-configuration"
+
+VMS=( $( virsh list --all --name ) )
+for vm in ${VMS[@]}; do
+  virsh reboot ${vm}
+  until virsh domstate ${vm} | grep "running"; do
+    echo "${vm} not yet running"
+    sleep 5
+  done
+done
+
+echo "Waiting for API before applying the rename and dnsmasq MachineConfigs..."
+until oc get clusterversion &> /dev/null
+do
+  echo "Waiting for API before applying the rename and dnsmasq MachineConfigs..."
+  sleep 5
+done
```

Comment:
Would it make sense to use `oc adm wait-for-stable-cluster` here as well?

Reply:
Oh yes, good point, I've done it when testing locally and forgot to copy it here :+1:

Reply:
Oh I apologize, although I have tried it locally, it's much faster to wait only for the API to be up, instead of the cluster report being healthy. Do you think it still makes sense to wait for the whole cluster to be stable?

Reply:
Arguably doing anything while the cluster is unstable might be a bad idea

Reply:
We don't need this one since we are getting rid of the [VM reboot](https://github.com/openshift/release/pull/44454#discussion_r1389599165). I'll just remove this loop as well.